indexing-service/src/main/java/org/apache/druid/indexing/common/task/CompactionTask.java

```diff
@@ -866,7 +866,7 @@ IndexTuningConfig computeTuningConfig(List<Pair<QueryableIndex, DataSegment>> qu
             nonNullTargetCompactionSizeBytes
         );
         return (tuningConfig == null ? IndexTuningConfig.createDefault() : tuningConfig)
-            .withMaxRowsPerSegment(maxRowsPerSegment);
+            .withMaxRowsPerSegment(maxRowsPerSegment).withMaxTotalRows(Long.MAX_VALUE);
```

Comment:
Would you please add a comment here why we set `maxTotalRows` to `Long.MAX_VALUE`? Probably it goes like `Setting maxTotalRows to Long.MAX_VALUE to respect the computed maxRowsPerSegment. If this is set to something too small, compactionTask can generate small segments which need to be compacted again, which in turn making auto compaction stuck in the same interval`.

Reply:
@jihoonson Yes, of course. I have added the comments. Thanks!